lib/puppet/type/user.rb

```diff
@@ -391,7 +391,7 @@ def insync?(current)
 
       validate do |value|
         if value.intern != :absent and value !~ /^\d{4}-\d{2}-\d{2}$/
-          raise ArgumentError, "Expiry dates must be YYYY-MM-DD or the string \"absent\""
+          raise ArgumentError, _("Expiry dates must be YYYY-MM-DD or the string \"absent\"")
```

Comment:
Do you think its worth a translator note that YYYY-MM-DD reflects a four digit year two digit month two digit day separated by dashes? Maybe that's self-evident, not sure how/if it translates though